### PR TITLE
Hero copy animation

### PR DESCRIPTION
**Pull request by zediogoviana**
Hero copy message animation.

Why:

- Necessary to implement bottom-up and up-bottom slide animation for hero copy message.

How:

- A timer to control which text to display and when to start a new animation.

- Used `Framer Motion` for animations.

### CODE REVIEW

File: src/components/HeroCopyElement/index.css [hunk 1]
```diff
@@ -27,12 +27,53 @@
...
+
 .message {
+  position: relative;
+
   display: flex;
   flex-direction: column;
   justify-content: space-between;
   width: $eight-columns;
 
+  /*animation-duration: 2.5s;
```
Comment (filipebfernandes) [inline src/components/HeroCopyElement/index.css:71]: you can remove the comments

File: src/components/HeroCopyElement/index.js [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
   function getClassName(place) {
     if (place === position) {
       return "line-klein-blue";
     }
-
     return "line-baby-blue";
   }
 
+  function display(value) {
+    if (position === value) {
```
Comment (filipebfernandes) [inline src/components/HeroCopyElement/index.js:26]: alternative with ternary operator `return position === value ? "block" : "none"`

**Comment by taniaprocha** [comment]
@zediogoviana it's better only ask for code review after design review is finished

**Comment by filipebfernandes** [comment]
@zediogoviana since you are changing the hero, please revert the wave direction. Now it is from right to left and should be from left to right.

**Comment by taniaprocha** [comment]
design review ✅

File: src/components/HeroCopyElement/index.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+      setSeconds(newSeconds);
+      setPosition(newSeconds % 3);
+    }, 4000);
+
+    return () => clearInterval(interval);
+  }, [seconds, position]);
+
+  return dimensions.width > breakpointMobile ? (
+    <>
```
Comment (taniaprocha) [inline src/components/HeroCopyElement/index.js:106]: You can have `renderMobile` and `renderTabletDesktop` methods to make code cleaner

File: src/components/HeroCopyElement/index.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+      <div styleName="copy-section">
+        <div styleName="lines">
+          <motion.div
```
Comment (taniaprocha) [inline src/components/HeroCopyElement/index.js:109]: Here you can reduce some code by creating a method called `renderLine` that returns line 0, 1 or 2

File: src/sections/HomeHero/common.css [hunk 5]
```diff
@@ -74,17 +77,27 @@ $Breakpoint-parallax: 768px;
...
+
+  width: 100%;
+  height: calc($wave-height * 2 + $spacing-8);
 
-  z-index: 2;
+  margin-top: calc($wave-height - 1px);
 
-  max-width: $six-columns;
+  background-color: $white;
+  /*border: 2px solid black;*/
```
Comment (taniaprocha) [inline src/sections/HomeHero/common.css:98]: Remove comment

File: src/components/HeroCopyElement/index.js [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+            type="button"
+            transition={transitionLine}
+            animate={animateLine(0)}
+            aria-label="clickable info 0"
+            styleName={getClassName(0)}
```
Comment (taniaprocha) [inline src/components/HeroCopyElement/index.js:114]: I'd use `classNames` here:

`classNames({
    "line-klein-blue": position === 0,
    "line-baby-blue": position !== 0
  })
`

**Comment by zediogoviana** [comment]
@filipebfernandes if you could also review the code, it is ready now.

File: src/components/HeroCopyElement/index.js [hunk 7]
```diff
@@ -1,76 +1,207 @@
...
-        <Typography color="klein-blue" weight="bold" variant="h2">
-          {textPlaceholder[position].title}
-        </Typography>
-      </div>
-    </div>
-  );
+  }, [seconds, position]);
+
+  function renderLine(transition, place) {
+    return (
```
Comment (taniaprocha) [inline src/components/HeroCopyElement/index.js:102]: Now I think you can put `animateLine` logic here

Comment (zediogoviana) [reply src/components/HeroCopyElement/index.js:102]: You mean put the `function animateLine()` inside?

Comment (taniaprocha) [reply src/components/HeroCopyElement/index.js:104]: Only the logic (if....). That logic is relative to line, isn't it?

Comment (zediogoviana) [reply src/components/HeroCopyElement/index.js:104]: Yes! I thought you meant the whole function, but yeah, makes sense only the ìf()`part!

**Comment by filipebfernandes** [comment]
@zediogoviana on smaller devices like the iPhone 5S the materialize ideas get cut out since it needs a new line.
Screenshot 2020-06-15 at 09 45 12

@taniaprocha not specific to this but could you reduce the wave on this section?

Screenshot 2020-06-15 at 09 47 30

**Comment by taniaprocha** [comment]
@filipebfernandes I'll add that issue to our basecamp and discuss it with Inês. Let's finish this PR and merge it.

**Comment by zediogoviana** [comment]
@filipebfernandes and @taniaprocha, the small screen cut out is fixed. I added an extra breakpoint for `width < 370px`, where the problem started occurring. I think everything is covered up now.

**Pull request merged**